onnx/defs/shape_inference.h

```diff
@@ -6,6 +6,13 @@
 
 namespace ONNX_NAMESPACE {
 
+class GraphInferencer {
+  // Perform inferencing on the graph contained in GraphInferencer.
+  // Returns the graph output types post-inferencing.
+  virtual std::vector<const TypeProto*> doInferencing(
```

Comment:
Thanks a lot for this change! Please also consider the case of using input data to do shape inference in this interface.

Reply:
Added.